lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp

```diff
@@ -152,6 +189,50 @@ class EltwiseBinaryOpConversionPattern
   }
 };
 
+// Matmul op conversion pattern
+//
+class MatmulOpConversionPattern
+    : public TTNNToEmitCBaseOpConversionPattern<ttnn::MatmulOp> {
+
+public:
+  MatmulOpConversionPattern(const TypeConverter &typeConverter,
+                            MLIRContext *context, PatternBenefit benefit = 1)
+      : TTNNToEmitCBaseOpConversionPattern<ttnn::MatmulOp>(typeConverter,
+                                                           context, benefit) {}
+
+  LogicalResult
+  matchAndRewrite(ttnn::MatmulOp matmulOp, ttnn::MatmulOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    llvm::SmallVector<Value, 2> operands;
+    operands.append({adaptor.getA(), adaptor.getB()});
+
+    // Erase DPS operand
+    //
+    // TODO (#1661): Once matmul op is not mismodelled as DPS, this can be
```

Comment:
With the next metal uplift, we will have a matmul version with DPS output:
https://github.com/tenstorrent/tt-metal/issues/15038